src/controllers/item.js

```diff
@@ -1,21 +1,62 @@
+const ItemModel = require('../models/item');
+
 module.exports = {
   getAllItems: (req, res) => {
     res.send('All items');
   },
-  getSingleItem: (req, res) => {
-    res.send('Single item');
+  getSingleItem: async (req, res) => {
+    try {
+      const itemId = req.params.id;
+      const item = await ItemModel.findById(itemId);
+      if (!item) throw new Error('There is no item with the provided id!');
+      res.json(item);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
-  getAvailableItems: (req, res) => {
-    res.send('Available items');
+  getAvailableItems: async (req, res) => {
+    try {
+      const availableItems = await ItemModel.find({ isAvailable: true });
+      if (availableItems.length <= 0)
+        throw new Error('There are no available items at the moment!');
+      res.json(availableItems);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
-  getFilteredItems: (req, res) => {
-    res.send('Filtered items');
+  getFilteredItems: async (req, res) => {
+    try {
+      const typeQuery = req.query.type;
+      if (!typeQuery) throw new Error('Type query is required!');
+      const filteredItems = await ItemModel.find({
+        type: typeQuery,
+        isAvailable: true,
+      });
+      if (filteredItems.length <= 0)
+        throw new Error(`There are no available items of type ${typeQuery}!`);
+      res.json(filteredItems);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
   },
   updateItem: (req, res) => {
     res.send('Item with the specified ID has been updated');
   },
-  addItem: (req, res) => {
-    res.send('New item has been added');
+  addItem: async (req, res) => {
+    try {
+      const { name, type, description, photo, count = 1, owner } = req.body;
+      await ItemModel.create({
```

Comment:
@AmmarBaki2 just a question here: why not directly to use req.body inside create like ItemModel.create({ req.body .... ?

Reply:
While reviewing mehmet's code I noticed that, it is way simpler than what I did 😅 
I will change it now. Thank you 😊